src/models/kafkaCluster.ts

```diff
@@ -1,40 +1,28 @@
 import { Data, type Require as Enforced } from "dataclass";
-import * as vscode from "vscode";
+import { MarkdownString, ThemeIcon, TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ConnectionType } from "../clients/sidecar";
 import { CCLOUD_CONNECTION_ID, IconNames, LOCAL_CONNECTION_ID } from "../constants";
 import { CustomMarkdownString } from "./main";
+import { ConnectionId, IResourceBase, isCCloud } from "./resource";
 
-/** Base class for all KafkaClusters, be they local or be they CCloud */
-export abstract class KafkaCluster extends Data {
-  abstract readonly connectionId: string;
-  abstract readonly isLocal: boolean;
-  abstract readonly isCCloud: boolean;
+/** Base class for all KafkaClusters */
+export abstract class KafkaCluster extends Data implements IResourceBase {
+  abstract connectionId: ConnectionId;
+  abstract connectionType: ConnectionType;
+  iconName: IconNames = IconNames.KAFKA_CLUSTER;
 
   abstract name: string;
-
   abstract environmentId: string | undefined;
 
   id!: Enforced<string>;
   bootstrapServers!: Enforced<string>;
-  uri!: Enforced<string>;
+  uri?: string;
 }
 
-/** A local Kafka cluster */
-export class LocalKafkaCluster extends KafkaCluster {
-  readonly connectionId: string = LOCAL_CONNECTION_ID;
-  readonly isLocal: boolean = true;
-  readonly isCCloud: boolean = false;
-  readonly environmentId: undefined = undefined;
-
-  // this is solely for display purposes so we don't have to check whether a resource is either a
-  // LocalKafkaCluster or CCloudKafkaCluster when generating a label for a tree/quickpick/etc item
-  readonly name: string = "Local";
-}
-
-/** A CCloud Kafka cluster */
+/** A Confluent Cloud {@link KafkaCluster} with additional properties. */
 export class CCloudKafkaCluster extends KafkaCluster {
-  readonly connectionId: string = CCLOUD_CONNECTION_ID;
-  readonly isLocal: boolean = false;
-  readonly isCCloud: boolean = true;
+  readonly connectionId: ConnectionId = CCLOUD_CONNECTION_ID as ConnectionId;
```

Comment:
Is this and similar casts necessary, given `CCLOUD_CONNECTION_ID` is defined as

`export const CCLOUD_CONNECTION_ID = CCLOUD_CONNECTION_SPEC.id! as ConnectionId;`

Reply:
Nope! Removed in https://github.com/confluentinc/vscode/pull/654/commits/c6b66970817b4b4d22e927f0993eb640267b9925